dev/NavigationView/NavigationView.cpp

```diff
@@ -616,6 +710,31 @@ bool NavigationView::IsRootItemsRepeater(const winrt::DependencyObject& element)
     return false;
 }
 
+bool NavigationView::IsRootGridOfFlyout(const winrt::DependencyObject& element)
+{
+    if (auto grid = element.try_as<winrt::Grid>())
+    {
+        return grid.Name() == c_flyoutRootGrid;
```

Comment:
object comparison by a name property seems dangerous. who's to say there isn't a grid with this name?

Reply:
This is a good point. Currently this is the method that is being used to determine if an Item is being displayed in a flyout. I'll open a separate issue to update this so that it does not depend on the visual tree in order to gain this information.

Reply:
@ojhad Can you create an issue and link it here ?

Reply:
https://github.com/microsoft/microsoft-ui-xaml/issues/2173